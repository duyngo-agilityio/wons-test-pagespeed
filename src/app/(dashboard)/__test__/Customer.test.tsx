import { render, waitFor } from '@testing-library/react';

// Pages
import CustomerListPage from '../customers/page';

jest.mock('@/components', () => ({
  CustomerDrawerWrapper: () => (
    <div data-testid="customer-drawer">Customer Drawer</div>
  ),
  TableSkeleton: () => <div data-testid="table-skeleton">Table Skeleton</div>,
}));

jest.mock('@/ui', () => ({
  CustomerList: () => <div data-testid="customer-list">Customer List</div>,
}));

describe('CustomerListPage render', () => {
  it('should render and match with snapshot', async () => {
    const { container } = render(<CustomerListPage />);

    await waitFor(() => {
      expect(container).toBeInTheDocument();
    });

    expect(container).toMatchSnapshot();
  });
});
