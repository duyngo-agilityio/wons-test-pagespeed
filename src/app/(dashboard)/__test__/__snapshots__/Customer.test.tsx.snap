// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomerListPage render should render and match with snapshot 1`] = `
<div>
  <main>
    <section>
      <div
        class="base:mb-6 md:mb-7.5 md:flex justify-between md:items-center"
      >
        <h2
          aria-level="2"
          class="font-bold font-dm-sans text-blue-800 dark:text-white text-5xl w-full"
        >
          Customer List
        </h2>
        <div
          data-testid="customer-drawer"
        >
          Customer Drawer
        </div>
      </div>
      <div
        data-testid="customer-list"
      >
        Customer List
      </div>
    </section>
  </main>
</div>
`;
